### ISSUE TITLE
PartialImport WOH fails if any of the tracks mentioned in the source SMIL are not in the source flavours

### ISSUE THREAD

**Issue by Arnei** [opencast/opencast#1918]
The PartialImport WOH processes tracks from its source flavours based on a given source SMIL.
Currently, the whole source SMIL is iterated to find matching source tracks, no matter the specified source-keys. This leads to exceptions being thrown when the corresponding track to a given SMIL-Id can not be found, for example by not including the flavor where the track resides. 

```
No track matching smil Track-id: 610998d9-36fc-43cb-8fdc-a1d8f49be846
```

**To Reproduce**
1. Create a workflow containing the PartialImport Operation. It should have *presenter/source* as one of its souce flavours.
2. Use the Rest Ingest API to create a mediapackage.
3. Add at least two tracks to the mediapackage, one with flavor *presenter/source*and one with flavor *presenter/usedForSomethingElse*
4. Ingest it and start the workflow you created
5. See error

**Environment Information**
 - Opencast version r/8.x and develop

**Comment by lkiesow** [comment]
Since it is basically an internal communication between ingest endpoint and workflow operation, does it really make sense to use SMIL at all? What we really need is just a list of the flavor (or other identifying mark for tracks) and the start position, don't we?

**Comment by Arnei** [comment]
I suppose defining tracks by flavor could work as well, but might be unintuitive, as you would probably have to do something like `source/presenter-1`, `source/presenter-2` and so on when ingesting. Something you have to do nowhere else afaik.

Also getting rid of SMIL is probably a much bigger task than what would be necessary to fix this issue. Might be worth a new issue :D

**Comment by Arnei** [comment]
Solved by #2316, closing
